torch/csrc/jit/script/compiler.h

```diff
@@ -1,17 +1,21 @@
 #pragma once
+#include <functional>
 #include <memory>
 #include <string>
 
 #include "torch/csrc/jit/ir.h"
+#include "torch/csrc/jit/pybind.h"
 #include "torch/csrc/jit/script/tree_views.h"
 
 namespace torch {
 namespace jit {
 namespace script {
 
+using ResolutionCallback = std::function<py::function(Graph*, std::string)>;
```

Comment:
Ideally, this callback and this entire compilation unit should not have a dependency on python. Instead the callback can look something like:
```
Called when the compiler encounters a function call 
struct Resolver {
   virtual std::vector<Value*> resolveCall(SourceRange location /*for error reporting*/, std::string name, Graph * g, std::vector<Value*> inputs) = 0;
   // in the future we might have:
   virtual Value* resolveIdentifier(SourceRange location, std::string name, Graph *g) = 0;
   // this may be used to turn python identifiers that resolve to constants into actual constants
   // a = 3;
  //  @script
   // def foo(b): return a + b
};
ResolutionCallback = std::function<std::vector<Value*>(Graph*, std::string, std::vector<Value*>)>;
```
Then the python related stuff can be isolated to the files that define the pybind bindings.